### ISSUE TITLE
imshow() subplots with shared axes generate unwanted white spaces

### ISSUE THREAD

**Issue by rycmak** [matplotlib/matplotlib#1789]
When plotting two (or more) subplots which share the same axes, I see large areas of white spaces within the plots as seen here:
![sharexImage](https://f.cloud.github.com/assets/3713541/199960/4b04380e-80b0-11e2-935e-273dc2fb74ef.png)

``` python
import matplotlib.pyplot as plt
import numpy as np

fig = plt.figure()
ax = fig.add_subplot(2, 1, 1)
ax.imshow(np.random.random((10,10)))
ax.autoscale(False)
ax2 = fig.add_subplot(2, 1, 2, sharex=ax, sharey=ax)
ax2.imshow(np.random.random((10,10)))
ax2.autoscale(False)
plt.show()
```

These white spaces persist even after `autoscale(False)`, `ax.set_xlim(0, 10)`, and/or `ax.set_xbound(0, 10)`.  One way to get rid of the white spaces is to set `aspect=auto` in `imshow()`, but this is not always ideal.
One could also set a `figsize` and manually adjust the axes to get the image contained exactly within the figure without white spaces, but resizing the figure would make the white spaces appear again.

Is there a way to correct for this behaviour?

**Comment by philippos86** [comment]
I had a similar issue and been searching first about the reason (really couldn't know why), and got here while searching for a solution. However, the solutions was by using the `matplotlib.axes.Axes.set_xlim` to set the dimensions of the plot, since I know beforehand the dimensions of my graph.
[matplotlib.axes.Axes.set_xlim](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.set_xlim.html#matplotlib.axes.Axes.set_xlim)

**Comment by efiring** [comment]
This is fixed by #10033.
![image_subplots](https://user-images.githubusercontent.com/85125/34732080-440447c0-f508-11e7-8fff-ac54cf8ca471.png)
